spec/models/storage_location_spec.rb

```diff
@@ -118,20 +118,34 @@
         it "gives informative errors" do
           storage_location = create(:storage_location, :with_items, item_quantity: 10, item: item, organization: @organization)
           expect do
-            storage_location.decrease_inventory(distribution_but_too_much.to_a).errors
+            storage_location.decrease_inventory(distribution_but_too_much.line_item_values).errors
           end.to raise_error(Errors::InsufficientAllotment)
         end
 
         it "does not change inventory quantities if there is an error" do
           storage_location = create(:storage_location, :with_items, item_quantity: 10, item: item, organization: @organization)
           starting_size = storage_location.size
           begin
-            storage_location.decrease_inventory(distribution.to_a)
+            storage_location.decrease_inventory(distribution.line_item_values)
           rescue Errors::InsufficientAllotment
           end
           storage_location.reload
           expect(storage_location.size).to eq(starting_size)
         end
+
+        context "when the item is does not exist" do
+          it "it raises insufficient allotment error if item does not exist" do
+            nonexistent_item = {item_id: nil, quantity: 1, name: "Nonexistent Item"}
+            storage_location = create(:storage_location, :with_items, item_quantity: 10, item: item, organization: @organization)
+            starting_size = storage_location.size
+            begin
+              storage_location.decrease_inventory([nonexistent_item])
+            rescue Errors::InsufficientAllotment
```

Comment:
This actually seems like the wrong error - it's not that there isn't enough of the item, it's that the item is invalid entirely.

Is there a reason this test was added? I'd rather not mess too much with inventory stuff because we're redoing a lot of it.

Reply:
Maybe its not needed if stuff is getting reworked, however, currently both the following lines in `storage_location` would pass the tests.

```ruby
inventory_item = inventory_items.find_by(item_id: item_hash[:item_id]) || inventory_items.build
inventory_item = inventory_items.find_by(item_id: item_hash[:item_id])
```

The current test cases do not cover the situation where `find_by` returned `nil`. I added that test case just in case.

Reply:
I get the reasoning, but I'd rather not pollute the current PR with additional unrelated changes. Can we remove this for now?